Add tag constants

pbricout opened the pull request:
## Context

This PR add tag numerical constants for frames that are currently supported by the library.

I have already defined constants for `ETCO` and `COMR` frames but did not realise they were not supported yet, so I did not include the definitions.
Also definitions for `TFLT` and `TMED` are a bit special, I did not include them in this PR.

I did not update the versioning yet, let me know if you want to do a release on this PR or not.

## Details

- use "Constants" as a namespace (not TagConstants as the tag prefix felt unnecessary we are in a tag library already), let me know if you feel that `TagConstants` would be preferred
- there is an unfortunate duplicate between the implementation file and the declaration file due to the fact that the implementation is Javascript and not Typescript
- use naming conventions widely used by the community, see for example
  https://google.github.io/styleguide/jsguide.html#naming
  https://developer.mozilla.org/en-US/docs/MDN/Community
  https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/
- handle APIC picture type in create
  -  handle the picture id in frame `create` instead of hardcoded value of 3
  - update the documentation
  - update the test

pbricout commented on src/ID3Definitions.js:
```diff
@@ -209,8 +209,79 @@ const ENCODINGS = [
+ *
+ * @see {@link https://id3.org/} for more information.
+ */
+const Constants = {
```
I have already defined constants for `ETCO` and `COMR` frames but did not realise they were not supported yet, so I did not include the definitions.
Also definitions for `TFLT` and `TMED` are a bit special, I did not include them in this PR.

Zazama replied:
How about you commit the ETCO and COMR constants and I will implement them tomorrow? Then everything would be in place

pbricout replied:
I pushed a commit with them.

Zazama commented on src/ID3Frames.js:
```diff
@@ -65,6 +65,11 @@ module.exports.APIC = {
                 }
             }
 
+            const Constants = ID3Definitions.Constants.AttachedPicture
+            const pictureType = data.type || {};
+            const pictureTypeId = pictureType.id === undefined
+                ? Constants.PictureType.FRONT_COVER : pictureType.id
+
```
I've been targeting node.js 10+ for backwards compatibility (which doesn't have optional chaining), so I think it's worth it to keep it like you commited

pbricout replied:
OK, thanks

pbricout commented:
Strangely enough, I noticed that `index.d.ts` is formatted with 3 spaces instead of 4, we should probably fix that in another PR.

Zazama commented:
Alright, your constants pull request and ETCO/COMR frames are now merged 🎉

pbricout commented:
Thanks for the merge.

You did not question the fact that I have chosen `Constants` instead of `TagConstants` (or `TagsConstants`).
TBH I am still a little unsure. Two considerations:
- theses constants are actually used in the `Tags` type so it could make sense to prefix with `Tags`
- if later we want to introduce `Constants` for the rest of the API it would conflict and we will need to find a different name which maybe be challenging

Any preferences or thoughts? This is still not too late to amend. Thanks

Zazama commented:
I have thought about the clash between the `Constants` name and possible collusions with further updates, but I thought they could be mixed in the end.

But it's probably cleaner to seperate such constants, then it would make more sense to rename the current `Constants` export.

I'd personally prefer the `TagConstants` name because it reads gramatically correct. Also, the `Tags` type is the collection of multiple tags but the constants are used in a specific tag.

There's no new release yet so it's still not a problem to rename it. Do you think `TagConstants` is fine?

pbricout commented:
Let's go with `TagConstants` then , I'll provide a PR to rename.

Zazama commented:
I've changed it in https://github.com/Zazama/node-id3/commit/8410149e5df7fab90405be72695a9e26ff00686e

pbricout commented:
> I've changed it in [8410149](https://github.com/Zazama/node-id3/commit/8410149e5df7fab90405be72695a9e26ff00686e)

OK, I also had a commit ready with exactly the same changes, it got dropped when I updated by fork, so all good, thank.
